src/java/org/apache/cassandra/locator/HttpSeedProvider.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.locator;
+
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.cassandra.locator.HttpServiceConnector.resolveMetadataUrl;
+import static org.apache.cassandra.locator.HttpServiceConnector.resolveRequestTimeoutMs;
+
+public class HttpSeedProvider implements SeedProvider
+{
+    private static final Logger logger = LoggerFactory.getLogger(HttpSeedProvider.class);
+    private static final Pattern NEW_LINES_PATTERN = Pattern.compile("\\n");
```

Comment:
The `System.lineSeparator()` is used in the `apiCall` method and here we expect the `\n` symbol, I think the tests only work because we are running. If someone tried to run it locally on Windows it would fail. I doubt anyone would do that, but the expectations need to be the same.

Reply:
I think this answers it https://github.com/apache/cassandra/pull/2988/files#r1537872100

We might also use `Pattern.compile("\" + System.lineSeparator())` or similar ...